Src/FluentAssertions/Xml/XDocumentAssertions.cs

```diff
@@ -116,6 +118,44 @@ public AndConstraint<XDocumentAssertions> NotBeEquivalentTo(XDocument unexpected
             return new AndConstraint<XDocumentAssertions>(this);
         }
 
+        /// <summary>
+        /// Asserts that document is valid by binary represented XSD schema.
+        /// </summary>
+        /// <param name="xsd">XDS schema.</param>
+        /// <param name="because">
+        /// A formatted phrase as is supported by <see cref="string.Format(string,object[])" /> explaining why the assertion
+        /// is needed. If the phrase does not start with the word <i>because</i>, it is prepended automatically.
+        /// </param>
+        /// <param name="becauseArgs">
+        /// Zero or more objects to format using the placeholders in <paramref name="because" />.
+        /// </param>
+        /// <returns>Instance of <see cref="AndConstraint{T}"/>.</returns>
+        public AndConstraint<XDocumentAssertions> BeValidByXsd(byte[] xsd, string because = "", params object[] becauseArgs)
+        {
+            var isValid = true;
+            var errorMessage = default(string);
+            var schemaSet = new XmlSchemaSet();
+
+            using var xsdFileStream = new MemoryStream(xsd);
+            using var reader = XmlReader.Create(xsdFileStream);
+
+            schemaSet.Add(string.Empty, reader);
+            schemaSet.Compile();
+
+            Subject.Validate(schemaSet, (o, e) =>
+            {
+                isValid = false;
+                errorMessage = e.Message;
+            });
+
+            Execute.Assertion
+                .BecauseOf(because, becauseArgs)
+                .ForCondition(isValid)
+                .FailWith("Given XML is invalid by XSD-schema. {0}{reason}.", errorMessage);
```

Comment:
My recommendation:
`Expected XML document to match XSD{reason}, but failed with {0}.`